pandas/core/groupby/ops.py

```diff
@@ -577,6 +583,14 @@ def _cython_operation(
         elif is_datetimelike and kind == "aggregate":
             result = result.astype(orig_values.dtype)
 
+        if (
```

Comment:
This isn't pretty but don't really know of a better way to make SeriesGroupBy return NA instead of NaN

Reply:
see #32984;
after we merge that you can just make a small mod to use the same

Reply:
merged #32894 if you want to refactor

Reply:
you need to use maybe_cast_result_dtype here; I don't actually think we need the else any longer

Reply:
A little confused, do you mean `maybe_cast_result` (`maybe_cast_result_dtype` returns a numpy int dtype itself)? When I try using `maybe_cast_result` though I just get a numpy array back

Reply:
look at the function which takes the name of the operation 'add' and the dtype and gives you back the casting dtype. which is eactly what we need to do here; you just need to modfy maybe_cast_result.

rather than have an else at all.

Reply:
This seems to be causing lots of test failures (e.g., also sending numpy arrays down this path raises when maybe_cast_result tries to access a _values attribute). Also seems we'd need to remove the check for result[0] being an instance of the original dtype because then maybe_cast_result doesn't try to convert NaN to a nullable integer, but then categorical tests end up failing. Seems worth refactoring but I'm honestly not sure how.

Reply:
i would like to fine a way of using maybe_cast_result, i don't mind have a very simpl elif here. but this is not good as written.

Reply:
Made a couple changes (some from another PR) that simplifies it a little